dlrover/python/master/node/dist_job_manager.py

```diff
@@ -1019,6 +1053,47 @@ def _relaunch_node(self, node: Node):
         self._job_context.update_job_node(node)
         self._scaler.scale(plan)
 
+    def _relaunch_node_group(self, node_group: int):
+        """
+        Relaunch all nodes in the node_group group, no matter it is
+        running or pending.
+        The group index should be different with all existing group,
+        so we use max_group_idx to keep it in record
+
+        """
+        self._job_context.max_group_idx += 1
```

Comment:
Also, we should not manipulate variables under the job context  this way(so implicitly). Should use a func instead.